### PR TITLE
Move gensym operations from `Symbol` to `Ident`

### PR DESCRIPTION
**Pull request by nnethercote**
Gensyms are always at the `Ident` level, and long-term we probably want to record gensym-ness in hygiene data.

r? @petrochenkov

### CODE REVIEW

File: src/libsyntax_pos/symbol.rs [hunk 1]
```diff
@@ -613,10 +613,12 @@ pub struct Ident {
 
 impl Ident {
     #[inline]
+    /// constructs a new identifier from a symbol and a span.
```
Comment (petrochenkov) [inline src/libsyntax_pos/symbol.rs]:
```suggestion
    /// Constructs a new identifier from a symbol and a span.
```

**Comment by petrochenkov** [comment]
r=me after rebase

**Comment by nnethercote** [comment]
@rust-timer build e67ebe6fdfb76cc820a7bc320c933953dec06067

**Comment by nnethercote** [comment]
It's a slight performance regression, but I think it's necessary to take at some point in order to move gensym-ness away from symbols. We've had a bunch of significantly larger performance improvements related to symbol handling recently, which more than make up for this.

@bors r=petrochenkov

**Pull request merged**